pkgs/top-level/all-packages.nix

```diff
@@ -7491,9 +7491,9 @@ in
 
   haskell = callPackage ./haskell-packages.nix { };
 
-  haskellPackages = dontRecurseIntoAttrs (haskell.packages.ghc865.override {
-    overrides = haskell.packageOverrides;
-  });
+  haskellPackages = dontRecurseIntoAttrs (haskell.packages.ghc865.override (old: {
+    overrides = lib.composeExtensions (old.overrides or (self: super: {})) haskell.packageOverrides;
```

Comment:
Is haskell.packageOverrides even necessary here? ghc865 should pick it up from:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/haskell-packages.nix#L24

Reply:
Not really sure, there might be some `.extend/.override/packageOverrides` whatever magic we're not seeing here, or maybe not. Unless somebody has a reason to keep this, I'll just remove this.

Reply:
Yeah actually you're right, this is handled already, I'll remove the ugliness here